### ISSUE TITLE
Add different unison modes

### ISSUE THREAD

**Issue by Jona-D** [surge-synthesizer/surge#3150]
I was wondering if it would be possible to add different unison modes, for example there's a mode called "Shimmer" in Ableton's wavetable synth that produces a "hypersaw" like effect.

**Here's a description from Ableton's manual**:

_"Shimmer: the oscillator pitches are jittered at random intervals, giving a shimmering reverb-like effect. A small amount of wavetable offset is also applied for extra fullness."_

**Also here's a video demonstrating what i'm referring to:**

https://youtu.be/BwIsnyZJfjc?t=58

**Comment by mkruselj** [comment]
Yeah it's possible ("it's just code"), but not in immediate plans for 1.8.

**Comment by baconpaul** [comment]
I like this idea and yeah 19 work next year will focus on oscillators more so let me tentatively tag this for then as we wrap up the 18 release

**Comment by mkruselj** [comment]
Also, apparently even voices are spread differently in stereo in other synths compared to Surge. See this example from Dune 3. Could be an additional option once we have unison spread control.

![even unison voices pan spread](https://user-images.githubusercontent.com/2393720/115724906-a63c2400-a381-11eb-9dfa-27412dd6d7d3.gif)

**Comment by baconpaul** [comment]
Yeah I think "unison panning strategy' as a parameter is interesting
It looks like what they are doing is: panning as if they have an odd number and just dropping the center voice.

**Comment by mkruselj** [comment]
Yeah I'm thinking it would be using "deform modes" on RMB of unison spread control once we have it.

**Comment by ryukra** [comment]
Yeah, this is a very good idea, especially the random feature that you have in surge/ableton.

**Comment by teolaz** [comment]
+1 :)

**Comment by mkruselj** [comment]
Here's description of unison detune options in Dune 3:

- Linear mode tunes all oscillators around the center with equal spacing.
- Nonlinear mode moves some oscillators closer to the center.
- Gaussian mode uses a Gaussian distribution for the volume of the oscillators.
- Alternate mode lowers every other oscillator in volume when changing the amount parameter.
- Random uses a random tuning for all oscillators whenever a new key is pressed.
- Perfect 5th is like Linear, except that every other oscillator is pitched up by +7 semitones
- Minor is like Linear, except that a minor chord is generated with 4 oscillators or more.
- Major is like Linear, except that a major chord is generated with 4 oscillators or more.
- Sub Osc is like Linear, except that every other oscillator is pitched down by -12 semitones.
- SWARM is a new oscillator stack model in DUNE 3, where all oscillators in the stack are modulated individually. This mode adds an extra RATE knob, which controls the rate of modulation.

As for controls, we will need 4:

- Unison Voices
- Unison Detune
- Unison Spread (stereo width)
- Unison Volume (adjusts the volume of non-center unison voices - there's one center voice with odd unison counts, and two with even counts)

The above detune options could go in context menu of Unison Detune parameter.


Thinking some more, I suppose we could split these options to two categories: those affecting the pitch, and those affecting the volume. As such, Gaussian and Alternate could be separated to the context menu of Unison Volume parameter (along with Linear option of course), and then it would apply to all other detune modes, which is one-upping Dune 3, actually. 🙂
